### PR TITLE
Feature/trucks api

### PR DESCRIPTION
**Pull request by Shakikka**
## What does this PR do?

- Hooks up backend truck api to the frontend

## How should it be tested?

- Login and look for the  trucks array in the console

## What do the future iterations hold?

- setting incoming data to state
- hooking new address up to backend

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -22,13 +24,21 @@ class App extends Component {
     const lat =  Number(user.data.attributes.lat)
     const lng = Number(user.data.attributes.long)
     const truckList = this.createLocationList(user)
-    this.setState({lat: lat, lng: lng, truckList:[...this.state.truckList, ...truckList ]})
+    this.setState({lat: lat, lng: lng, truckList:[...this.state.truckList, ...truckList ], trucks:[...this.state.trucks, ...user.data.attributes.trucks]})
```
Comment (elizhahn) [inline src/components/App.js:27]: You can get rid of this method now and update the lat and lng state from the fetch call upon login

File: src/components/App.js [hunk 2]
```diff
@@ -22,13 +24,21 @@ class App extends Component {
...
     fetchUserName(userName)
-    .then(data => console.log(data))
-    .catch(error => console.log("error"))
+    .then(data => {
+      console.log('id', data.data.id)
+      const id = data.data.id
+      fetchTrucks(id)
+      .then(trucks => {
+        console.log('trucks', trucks.data)
+        this.setState({ trucks: trucks.data})
```
Comment (elizhahn) [inline src/components/App.js:38]: you might need to pass the data from the first call into this fetch call to be able to use it

**Review by elizhahn — CHANGES_REQUESTED** [review]

File: src/components/App.js (hunk 2 above)
Comment (Shakikka) [reply src/components/App.js:38]: other than the id that we are passing into it?

File: src/components/App.js (hunk 1 above)
Comment (Shakikka) [reply src/components/App.js:27]: We aren't receiving lat / long for the user from the backend currently.

Comment (elizhahn) [reply src/components/App.js:27]: ok we will need to have the backend team update that return

File: src/components/App.js (hunk 2 above)
Comment (elizhahn) [reply src/components/App.js:38]: nevermind on this, we need to have the lat and lng being returned from the api call

**Pull request merged**